tests/unit/samplegen/test_template.py

```diff
@@ -301,6 +312,226 @@ def test_dispatch_map_loop():
+    )
+
+
+def test_render_calling_form_longrunning():
+    check_template(
+        '''
+        {% import "feature_fragments.j2" as frags %}
+        {% set callingFormEnum  = {
+                           "Request": "request",
+                           "RequestPagedAll": "requestpagedall",
+                           "RequestPaged": "requestpaged",
+                           "RequestStreamingServer": "requeststreamingserver",
+                           "RequestStreamingBidi": "requeststreamingbidi",
+                           "LongRunningRequestPromise": "longrunningrequestpromise"} %}
+        {{ frags.renderCallingForm("TEST_INVOCATION_TXT", "longrunningrequestpromise",
+                                   callingFormEnum,
+                                   [{"print": ["Test print statement"]}]) }}
+        ''',
+        '''
+        operation = TEST_INVOCATION_TXT
+        
+        print("Waiting for operation to complete...")
+
+        response = operation.result()
+        print("Test print statement")
+
+        '''
+
+    )
+
+
+def test_render_method_call_basic():
+    # The callingForm and callingFormEnum parameters are dummies,
+    # which we can get away with because of duck typing in the template.
+    check_template(
+        '''
+        {% import "feature_fragments.j2" as frags %}
+        {{ frags.renderMethodCall({"rpc": "Categorize", "request": [{"base": "video"},
+                                                                    {"base": "audio"},
+                                                                    {"base": "guess"}]},
+                                  "result", {"RequestStreamingBidi": "bidi",
+                                             "RequestStreamingClient": "client"}) }}
+        ''',
+        "\nclient.Categorize(video, audio, guess)\n"
+    )
+
+
+def test_render_method_call_bidi():
+    # The callingForm and callingFormEnum parameters are dummies,
+    # which we can get away with because of duck typing in the template.
+    check_template(
+        '''
+        {% import "feature_fragments.j2" as frags %}
+        {{ frags.renderMethodCall({"rpc": "Categorize", "request": [{"base": "video"}]},
+                                  "bidi", {"RequestStreamingBidi": "bidi",
+                                             "RequestStreamingClient": "client"}) }}
+        ''',
+        "\nclient.Categorize([video])\n"
+    )
+
+
+def test_render_method_call_client():
+    # The callingForm and callingFormEnum parameters are dummies,
+    # which we can get away with because of duck typing in the template.
+    check_template(
+        '''
+        {% import "feature_fragments.j2" as frags %}
+        {{ frags.renderMethodCall({"rpc": "Categorize", "request": [{"base": "video"}]},
+                                  "client", {"RequestStreamingBidi": "bidi",
+                                             "RequestStreamingClient": "client"}) }}
+       ''',
+        "\nclient.Categorize([video])\n"
```

Comment:
Should it be `categorize` here? I _think_ public methods in Python gapics are in `snake_case`.

Reply:
You're right: both the existing samples from https://github.com/beccasaurus/gapic-docs-samples and the output of the microgenerator use snake case. Will address.